AssignResourceChangeStatus.cs

```diff
@@ -93,17 +93,26 @@ private IssueDto UpdateStatus( IssueDtoEventArgs args )
             //If there is no resource, then do not want to do anything
             if ( args.Issue.Entity.GetResources().Count == 0 ) return args.Issue;
 
+            //if there are already resources assigned, then do not want to do anything
+            if ( args.Previous != null && args.Previous.Entity != null && args.Previous.Entity.GetResources().Count > 0 ) return args.Issue;
+
             var project = args.Context.Projects.Get( args.Issue.Entity.ProjectId );
 
             if ( project == null ) return args.Issue;
 
             //If the status is NOT the default for the project/type AND default statusId has been selected then exit
-            ProjectManager pm = GeminiApp.GetManager<ProjectManager>( args.User ); ;
-            var projectDefaults = pm.GetDefaults( args.Issue.Project, args.Issue.Entity.TypeId );
-            
-            //          Current status is not the default statusId           and   there was a default status set
-            if( args.Issue.Entity.StatusId != projectDefaults.Values.StatusId && projectDefaults.Values.StatusId  > 0 )
+            ProjectManager projectManager = GeminiApp.GetManager<ProjectManager>( args.User );
+            var projectDefaults = projectManager.GetDefaults( args.Issue.Project, args.Issue.Entity.TypeId );
+
+            if (projectDefaults == null)
```

Comment:
So the app won't work if you don't have project default?

Reply:
That is what was agreed, though in actual fact, Project Defaults is never null, it will just be a new ProjectDefault object (which then exits because there is no default status id). The null check was here in case that implementation ever changed to return null if there were no project defaults at all, rather than the empty ProjectDefaults class which is now.

Reply:
Yes, but even so we return a new "defaults" object which will have a 0 status id thus the app won't work.